pilot/pkg/serviceregistry/serviceentry/controller_test.go

```diff
@@ -714,7 +729,9 @@ func TestServiceDiscoveryWorkloadUpdate(t *testing.T) {
 		}
 		expectProxyInstances(t, sd, instances, "4.4.4.4")
 		expectServiceInstances(t, sd, dnsSelector, 0, instances)
-		expectEvents(t, events, Event{kind: "xds"})
+		expectEvents(t, events,
+			Event{kind: "edscache", host: "dns.selector.com", namespace: dnsSelector.Namespace},
```

Comment:
Why are we updating edscache for a DNS service entry? edscache is for EDS which isn't used for DNS service entries?

Is this intended or a side effect?

Reply:
It  is intended for  serviceentry with selector

Reply:
This is not necessary, but i found we didnot distinguish DNS address since very long before. So we always  update eds for them as well